mock_hardware/include/mock_components/generic_system.hpp

```diff
@@ -100,6 +99,37 @@ class HARDWARE_INTERFACE_PUBLIC GenericSystem : public hardware_interface::Syste
   std::vector<std::vector<double>> gpio_commands_;
   std::vector<std::vector<double>> gpio_states_;
 
+  double actuator_slowdown_;
+
+  // used for the Transmission pass through.
+  // read: actuator_interface.state_->Transmission->joint_interface.state_
+  // write: joint_interface.command_->Transmission->actuator_interface.command_
+  // And StateInterface(joint_interface.state_)
+  struct InterfaceData
+  {
+    // TODO(Manuel) set initial to NaN and on_init initialize to given value in info or 0.0
+    explicit InterfaceData(const std::string & name)
+    : name_(name),
+      command_(0.0),  // command_(std::numeric_limits<double>::quiet_NaN()),
+      state_(0.0),    // state_(std::numeric_limits<double>::quiet_NaN()),
+      transmission_passthrough_(
+        0.0)  // transmission_passthrough_(std::numeric_limits<double>::quiet_NaN())
+    {
+    }
+
+    std::string name_;
+    double command_;
+    double state_;
+    // this is the "sink" that will be part of the transmission Joint/Actuator handles
+    double transmission_passthrough_;
+  };
```

Comment:
I am not sure about this structure. but also it looks like something that might be added to `.cpp` file. Especially, since we have this game with `0.0` and `quiet_NaN()`.

I have to check the implemenation to tell more about it.

Reply:
what about the struct? need it in the .hpp since we define the lists with it